pkg/agent/agent.go

```diff
@@ -313,6 +304,8 @@ func (i *Initializer) Initialize() error {
 	wg.Add(1)
 	// routeClient.Initialize() should be after i.setupOVSBridge() which
 	// creates the host gateway interface.
+	i.networkConfig.AgentFeatures.IPv4Enabled = config.IsIPv4Enabled(i.nodeConfig, i.networkConfig.TrafficEncapMode)
```

Comment:
1. I feel IPv4Enabled and IPv6Enabled should not be in AntreaFeatures, as they are not configured through feature settings in Agent config.
2. Probably move the initialization to line 283 after initNodeLocalConfig.

Reply:
For 1), do you mean moving IPv4Enabled/IPv6Enabled to NetworkConfig fields directly? @jianjuns

Reply:
Yes, I feel they probably should be in NetworkConfig. Do you agree they are different from other fields in AgentFeatures?

Reply:
Actually, I think it is a bit confusion about IPv4Enabled/IPv6Enabled. These flags are proposed because we need them for IPv6 only and dual-stack  case, and dual-stack is actually a feature gate but not in Antrea Agent. So I think both are OK to place it in AgentFeatures or NetworkConfig directly. I originally chose to place it under AgentFeatures because I wanted to use a single struct for Agent flags.

Reply:
Besides, since I am working on VM/BM code change, in this scenarion we also need IPv4/IPv6Enabled flags for datapath, but we actually don't need other fields in NetworkConfig. This is another reason I originally chose to place them under AgentFeatures.

Reply:
Yeah, but if AntreaFeatures are for features, IPv4Enabled/IPv6Enabled do not fit there.

Reply:
Agree, it is a bit wired to place them there.

Reply:
Then I would move them to NetworkConfig in this PR.